gui/src/2D/controls/control.ts

```diff
@@ -1122,6 +1122,17 @@ export class Control {
 
         this._isEnabled = value;
         this._markAsDirty();
+        // if any child elements are under the pointer, we need to fire a pointerOut event on
+        if (!this._isEnabled) {
+            [this, ...this.getDescendants()].forEach(desc => {
```

Comment:
I'm afraid this will add pressure on the GC. Can we try to find a way to not create an array each time?

Reply:
Sure. How's this?